### ISSUE TITLE
Add Feature - Different Options to sort feed by (location, timestamp, relevance)

### ISSUE THREAD

**Issue by DevonLattrell** [FightPandemics/FightPandemics#1303]
Links to Screen Shots of Wireframe view of feature below. This feature is for the feed to be sorted by location (near to far), post timestamp (new to old), most liked post (most to least), trending (most comments and shares to least). 

Ideally the Sort Search tab is a click button that leads to the drop down to exclusive sorting by category. (see Sort Search top right)

![Sort Search FP](https://user-images.githubusercontent.com/68662812/88967996-996c2300-d27c-11ea-87a6-01efad546c69.PNG)
![Sort Search Drop down FP](https://user-images.githubusercontent.com/68662812/88968007-a1c45e00-d27c-11ea-8e18-678cdb84043b.PNG)
![Sort Search Selection](https://user-images.githubusercontent.com/68662812/88968018-a852d580-d27c-11ea-98a7-2c004da274e6.PNG)

**Comment by rabbi08** [comment]
Currently testing on the POST displayed in STG, I have observed that there is NO particular sort criteria NON-LOGGED in user. The POST should be sorted as per 'Location: Near to Far' as per current expected functionality, but currently it looks that there is no sort criteria. Refer below GIF.

https://share.getcloudapp.com/YEuolpgO

Was about to log a bug but then found this issue during search for duplicate, hence adding my observation here. This looks great, only suggestion wanna add that for non-logged in user during onboarding, the default sorting for the Post should be as per location-near to far.

**Comment by joshmorel** [comment]
@rabbi08 - when you mean sorting by location for non-logged user are you thinking of something like this  #376 ?

**Comment by rabbi08** [comment]
> @rabbi08 - when you mean sorting by location for non-logged user are you thinking of something like this #376 ?

Yes, it should be filtering the post displayed, as per the location for the onboarding user.

**Comment by joshmorel** [comment]
Just to be clear @rabbi08 this was when you clicked "Show me postings from anywhere" when going through onboarding. Since the user isn't supplying a location we'd need to use another external API (not sure if there is a free one...). 

I think we should tackle this separately from this ticket.

**Comment by manualzuru** [comment]
@Yu-Alexa-Zhao I guess this is the initial design and not the final one as it looks different from our drowdown menus right?

**Comment by DevonLattrell** [comment]
@manualzuru this is definitely initial look and feel. I just wanted to wireframe my thought process.

**Comment by manualzuru** [comment]
Thanks @DevonLattrell makes sense✌🏽. @Naraujo13 please take a look.

**Comment by yu-zhao-xyz** [comment]
Yes we prototyped this feature before but took it out in MVP. Is this a V1.2 feature?

**Comment by Naraujo13** [comment]
Agree with @joshmorel here. 

Currently, posts are sorted by creation date when user is not logged since we didn't have a location in those cases. 
Ideally, we could use an external service to get it by IP, but that is not that straightforward (not sure about free services with this either), and I think a separate issue like #376 is better.

About having different sorting options, it makes sense and it's pretty straightforward to implement on the backend.
`Nearest`, `Recent` and `Most Liked` seem pretty straightforward, we just need to define the expected behavior for `Trending` sorting (define what is trending and how to calculate that).

E.G. trending is defined by number of comments in the last week or number of comments + likes

Depending on this definition it may be easy or harder to implement (comments are okay, likes are harder since we don't store the timestamp of the like, so we would need to change that also).

**Comment by manualzuru** [comment]
Let's start with the easiest then @Naraujo13 @DevonLattrell @joshmorel and then we can think about the trending one.

@Yu-Alexa-Zhao we decided to instead of shipping releases, we will ship continuously every two weeks the features that were added.

**Comment by DevonLattrell** [comment]
@Naraujo13 @joshmorel 
Seems like for us at the moment **Comment** trends is best due to time stamp...I am not sure if the **Share** feature would help add to the trend. If the easiest and shortest way to value is showing `Whats Hot` or `Trending` through **Comments** I would begin with that and evolve the definition later. 

I will check up on how other feeds define `Trending` and check back in.

**Comment by manualzuru** [comment]
Sounds good to me @DevonLattrell

**Comment by rabbi08** [comment]
> Just to be clear @rabbi08 this was when you clicked "Show me postings from anywhere" when going through onboarding. Since the user isn't supplying a location we'd need to use another external API (not sure if there is a free one...).
> 
> I think we should tackle this separately from this ticket.

Thnx @joshmorel ,I have logged #1323 for proper tracking of this scenario.

**Comment by DevonLattrell** [comment]
@Naraujo13 @joshmorel 
After some review, the time stamping is the most important thing for the `Trending` sort to function as expected. Ideally in the future we may have time stamps for the `Likes` on a post in order to fully encompass the sort. 

I propose we start with a 'Trending' definition of posts with the most comments over a 48 hour period. Let me know what you guys think.
